rare/components/dialogs/uninstall_dialog.py

```diff
@@ -9,61 +8,70 @@
+        self.setWindowTitle(f'{header} "{rgame.app_title}" - {QCoreApplication.instance().applicationName()}')
         self.info_text = QLabel(
-            self.tr("Do you really want to uninstall <b>{}</b>?").format(game.app_title)
+            self.tr("Do you really want to uninstall <b>{}</b>?").format(rgame.app_title)
         )
-        layout.addWidget(self.info_text)
+
         self.keep_files = QCheckBox(self.tr("Keep game files."))
+        self.keep_files.setChecked(bool(options.keep_files))
         self.keep_config = QCheckBox(self.tr("Keep game configuation."))
-        form_layout = QVBoxLayout()
-        form_layout.setContentsMargins(6, 6, 0, 6)
-        form_layout.addWidget(self.keep_files)
-        form_layout.addWidget(self.keep_config)
-        layout.addLayout(form_layout)
+        self.keep_config.setChecked(bool(options.keep_config))
 
-        button_layout = QHBoxLayout()
-        self.ok_button = QPushButton(
+        self.uninstall_button = QPushButton(
             icon("ei.remove-circle", color="red"), self.tr("Uninstall")
         )
-        self.ok_button.clicked.connect(self.ok)
+        self.uninstall_button.setObjectName("UninstallButton")
+        self.uninstall_button.clicked.connect(self.__on_uninstall)
 
         self.cancel_button = QPushButton(self.tr("Cancel"))
-        self.cancel_button.clicked.connect(self.cancel)
+        self.cancel_button.clicked.connect(self.__on_cancel)
 
-        button_layout.addWidget(self.ok_button)
-        button_layout.addStretch(1)
+        form_layout = QVBoxLayout()
+        form_layout.setContentsMargins(-1, -1, 0, -1)
+        form_layout.addWidget(self.keep_files)
+        form_layout.addWidget(self.keep_config)
+
+        button_layout = QHBoxLayout()
         button_layout.addWidget(self.cancel_button)
+        button_layout.addStretch(1)
+        button_layout.addWidget(self.uninstall_button)
+
+        layout = QVBoxLayout()
+        layout.addWidget(self.info_text)
+        layout.addLayout(form_layout)
         layout.addLayout(button_layout)
+
         self.setLayout(layout)
 
-        if get_sdl_appname(game.app_name) is not None:
+        if get_sdl_appname(rgame.app_name) is not None:
             self.keep_config.setChecked(True)
 
-        self.options: Tuple[bool, bool, bool] = (
-            False, self.keep_files.isChecked(), self.keep_config.isChecked()
-        )
+        self.options: UninstallOptionsModel = options
 
-    def get_options(self) -> Tuple[bool, bool, bool]:
-        self.exec_()
-        return self.options
+    def closeEvent(self, a0: QCloseEvent) -> None:
+        self.result_ready.emit(self.options)
+        super(UninstallDialog, self).closeEvent(a0)
 
-    def ok(self):
-        self.options = (True, self.keep_files.isChecked(), self.keep_config.isChecked())
+    def __on_uninstall(self):
+        self.options.values = (True, self.keep_files.isChecked(), self.keep_config.isChecked())
         self.close()
```

Comment:
I am trying to avoid double signal connections due to https://doc.qt.io/qt-6/qmetaobject.html#connectSlotsByName. AFAIU the double underscores are going to avoid it, but for anyone knowing Qt, a better name might indicate the opposite. I think the suggestion is good though and my fears are unfounded.